wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServerChecker.java

```diff
@@ -8,7 +8,7 @@
  * Contributors:
  *   Codenvy, S.A. - initial API and implementation
  *******************************************************************************/
-package org.eclipse.che.workspace.infrastructure.docker.server;
+package org.eclipse.che.api.workspace.server.hc;
```

Comment:
I find it a bit weird that these components are part of the workspace api module.
I clearly understand that spi model is defined under package `org.eclipse.che.api.workspace.spi`, so you need a place probably somewhere near the spi definition.

Consider the following module structure
```
--wsmaster
  |--che-workspace-spi
     |--model
     |--hc
     |--bootstrapper
  |--che-workspace-api
```
It allows you to place all the spi related stuff in the same module, which i think is good.
Clearly you might not want to do it under this PR, but what do you think the structure?
@garagatyi, @sleshchenko, @skabashnyuk what do you guys think?